src/sql/parts/connection/common/connectionActions.ts

```diff
@@ -17,45 +17,99 @@ import { IConnectionManagementService } from 'sql/parts/connection/common/connec
+		this.enabled = true;
+	}
 
-		public run(): TPromise<void> {
-			let self = this;
-			return self.promptToClearRecentConnectionsList().then(result => {
+	public set useConfirmationMessage(value: boolean) {
+		this._useConfirmationMessage = value;
+	}
+
+	public run(): TPromise<void> {
+		if (this._useConfirmationMessage) {
+			return this.promptConfirmationMessage().then(result => {
 				if (result) {
-					self._connectionManagementService.clearRecentConnectionsList();
-					self._messageService.show(Severity.Info, nls.localize('ClearedRecentConnections', 'Recent connections list cleared'));
+					this._connectionManagementService.clearRecentConnectionsList();
+					this._onRecentConnectionsRemoved.fire();
+				}
+			});
+		} else {
+			return this.promptQuickOpenService().then(result => {
+				if (result) {
+					this._connectionManagementService.clearRecentConnectionsList();
+					this._messageService.show(Severity.Info, nls.localize('ClearedRecentConnections', 'Recent connections list cleared'));
+					this._onRecentConnectionsRemoved.fire();
 				}
 			});
 		}
+	}
 
-		private promptToClearRecentConnectionsList(): TPromise<boolean> {
-			const self = this;
-			return new TPromise<boolean>((resolve, reject) => {
-				let choices: { key, value }[] = [
-					{ key: nls.localize('connectionAction.yes', 'Yes'), value: true },
-					{ key: nls.localize('connectionAction.no', 'No'), value: false }
-				];
-
-				self._quickOpenService.pick(choices.map(x => x.key), { placeHolder: nls.localize('ClearRecentlyUsedLabel', 'Clear Recent Connections List'), ignoreFocusLost: true }).then((choice) => {
-					let confirm = choices.find(x => x.key === choice);
-					resolve(confirm && confirm.value);
-				});
+	private promptQuickOpenService(): TPromise<boolean> {
+		const self = this;
+		return new TPromise<boolean>((resolve, reject) => {
+			let choices: { key, value }[] = [
+				{ key: nls.localize('connectionAction.yes', 'Yes'), value: true },
+				{ key: nls.localize('connectionAction.no', 'No'), value: false }
+			];
+
+			self._quickOpenService.pick(choices.map(x => x.key), { placeHolder: nls.localize('ClearRecentlyUsedLabel', 'Clear Recent Connections List'), ignoreFocusLost: true }).then((choice) => {
+				let confirm = choices.find(x => x.key === choice);
+				resolve(confirm && confirm.value);
 			});
-		}
+		});
+	}
+
+	private promptConfirmationMessage(): TPromise<boolean> {
+		let confirm: IConfirmation = {
+			message: nls.localize('clearRecentConnectionMessage', 'Are you sure you want to delete all the connections from the list?'),
+			primaryButton: nls.localize('connectionDialog.yes', 'Yes'),
+			secondaryButton: nls.localize('connectionDialog.no', 'No'),
+			type: 'question'
+		};
+
+		// @SQLTODO
+		return new TPromise<boolean>((resolve, reject) => {
+			let confirmed: boolean = this._messageService.confirm(confirm);
+			resolve(confirmed);
+		});
+
+		//this._messageService.confirm(confirm).then(confirmation => {
```

Comment:
We can get rid of the commented code, while we're at it.

Reply:
Did @abist  or @kburtram  comment this? Is it safe to remove?

Reply:
I didn't comment it, however, it's not being used anyway, and we just updated this side of the code. So, it should be safe to remove the commented parts.